File: api/routes/comparison.py
from fastapi import APIRouter, HTTPException
from typing import List, Dict, Any

router = APIRouter(prefix="/api/v1/teams", tags=["comparison"])

@router.get("/comparison")
def get_teams_comparison() -> List[Dict[str, Any]]:
    from clickhouse.client import execute
    
    teams_data = []
    try:
        result = execute("""
            SELECT 
                toString(team_id) as team_id,
                source_type,
                event_type,
                count() as cnt
            FROM events
            WHERE occurred_at > now() - INTERVAL 7 DAY
            GROUP BY team_id, source_type, event_type
            ORDER BY team_id, source_type
        """)
        teams_data = result if result else []
    except Exception as e:
        print(f"Comparison query error: {e}")
        return []
    
    teams: Dict[str, Dict[str, Any]] = {}
    for row in teams_data:
        tid, source, etype, cnt = str(row[0]), row[1], row[2], row[3]
        if tid not in teams:
            teams[tid] = {"team_id": tid, "prs": 0, "tasks": 0, "ci_runs": 0}
        if source == "git" and etype in ("pr_opened", "pr_merged"):
            teams[tid]["prs"] += cnt
        elif source == "pm" and etype in ("task_created", "task_completed"):
            teams[tid]["tasks"] += cnt
        elif source == "cicd":
            teams[tid]["ci_runs"] += cnt
    
    return list(teams.values())

File: internal/pkg/biz/comparison.go
package biz

import (
	"context"
)

type TeamComparison struct {
	TeamID   string `json:"team_id"`
	Name     string `json:"name"`
	PRs      int    `json:"prs"`
	Tasks    int    `json:"tasks"`
	CIRuns   int    `json:"ci_runs"`
}

type ComparisonService struct {
	db DatabaseQuery
}

func NewComparisonService(db DatabaseQuery) *ComparisonService {
	return &ComparisonService{db: db}
}

func (s *ComparisonService) Get(ctx context.Context) ([]TeamComparison, error) {
	rows, err := s.db.Query(ctx, `
		SELECT t.id as team_id, t.name,
			(SELECT count() FROM events WHERE team_id = t.id AND source_type = 'git' AND event_type = 'pr_merged' AND occurred_at > now() - INTERVAL 7 DAY) as prs,
			(SELECT count() FROM events WHERE team_id = t.id AND source_type = 'pm' AND event_type = 'task_completed' AND occurred_at > now() - INTERVAL 7 DAY) as tasks,
			(SELECT count() FROM events WHERE team_id = t.id AND source_type = 'cicd' AND event_type = 'pipeline_run' AND occurred_at > now() - INTERVAL 7 DAY) as ci_runs
		FROM teams t
	`)
	if err != nil {
		return nil, err
	}

	var teams []TeamComparison
	for _, r := range rows {
		teams = append(teams, TeamComparison{
			TeamID: getString(r["team_id"]),
			Name:   getString(r["name"]),
			PRs:    int(getInt64(r["prs"])),
			Tasks:  int(getInt64(r["tasks"])),
			CIRuns: int(getInt64(r["ci_runs"])),
		})
	}

	return teams, nil
}